File: TeacherTracker.Api/Models/TrackingNote.cs
namespace TeacherTracker.Api.Models;

/// A timestamped observation about a student (behavior, progress, etc.).
public class TrackingNote
{
    public int Id { get; set; }
    public string Category { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }

    public int StudentId { get; set; }
    public Student? Student { get; set; }
}

File: teacher_tracker_app/lib/models/tracking_note.dart
class TrackingNote {
  const TrackingNote({
    required this.id,
    required this.category,
    required this.content,
    required this.createdAt,
    required this.studentId,
  });

  final int id;
  final String category;
  final String content;
  final DateTime createdAt;
  final int studentId;

  factory TrackingNote.fromJson(Map<String, dynamic> json) => TrackingNote(
        id: json['id'] as int,
        category: json['category'] as String? ?? '',
        content: json['content'] as String? ?? '',
        createdAt: DateTime.parse(json['createdAt'] as String).toLocal(),
        studentId: json['studentId'] as int,
      );

  static Map<String, dynamic> writeJson({
    required String category,
    required String content,
  }) =>
      {'category': category, 'content': content};
}
